### PR TITLE
[Internal] Unittest: Adds Coverage to validate Cosmos .NET SDK versioning strategy

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Unit test coverage to validate Cosmos .NET SDK versioning strategy

Versioning strategy: https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/docs/versioning.md

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcementTests.cs [hunk 1]
```diff
@@ -46,6 +49,31 @@ public void TelemetryContractChanges()
...
+            Trace.TraceInformation($"Official Version: {officialVersionText}");
+            Trace.TraceInformation($"Preview Version: {previewVersionText}");
+
+            Version officialVersion = new Version(officialVersionText);
+            Version previewVersion = new Version(previewVersionText);
+
+            // Preview minor version should always be one ahead of the official version
+            Assert.AreEqual(officialVersion.Major, previewVersion.Major, $"{officialVersionText} {previewVersionText}");
+            Assert.AreEqual(officialVersion.Minor + 1, previewVersion.Minor, $"{officialVersionText} {previewVersionText}");
+            Assert.AreEqual(officialVersion.Build, previewVersion.Build, $"{officialVersionText} {previewVersionText}");
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcementTests.cs:76]: This last assert seems wrong. https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/docs/versioning.md

When for example, 3.41.1 is released, we release 3.42.0-preview.1.

The Build would not match

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcementTests.cs:76]: That's true, this test doesn't cover the real package names but just the required changes in the `Directory.Build.props` file.

`-preview' is part of the build process. 

Goal: Is to guard about versioning changes developers do part of release process (Not end to end)

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcementTests.cs:76]: Offline synced-up, addressing the gap

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcementTests.cs [hunk 2]
```diff
@@ -46,6 +49,71 @@ public void TelemetryContractChanges()
...
+            XmlDocument xmldoc = new XmlDocument();
+            xmldoc.LoadXml(File.ReadAllText(VersionFile));
+
+            XmlNode propertyGroupNode = xmldoc.DocumentElement["PropertyGroup"];
+
+            string officialVersionText = propertyGroupNode?["ClientOfficialVersion"]?.InnerText;
+            string previewVersionText = propertyGroupNode?["ClientPreviewVersion"]?.InnerText;
+            string previewSuffixText = propertyGroupNode?["ClientPreviewSuffixVersion"]?.InnerText;
+
+            Trace.TraceInformation($"Official Version: {officialVersionText}");
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcementTests.cs:68]: nit: Prefer Logger. Logger outputs also to VSTest UI

**Review by ealsur — APPROVED** [review]

**Review by NaluTripician — APPROVED** [review]

**Pull request merged**